pkg/webhook/clusterresourceplacement/v1beta1_clusterresourceplacement_validating_webhook.go

```diff
@@ -42,15 +48,19 @@ func (v *clusterResourcePlacementValidator) Handle(_ context.Context, req admiss
 			klog.ErrorS(err, "failed to decode v1beta1 CRP object for create/update operation", "userName", req.UserInfo.Username, "groups", req.UserInfo.Groups)
 			return admission.Errored(http.StatusBadRequest, err)
 		}
-		if err := validator.ValidateClusterResourcePlacement(&crp); err != nil {
-			klog.V(2).InfoS("v1beta1 cluster resource placement has invalid fields, request is denied", "operation", req.Operation, "namespacedName", types.NamespacedName{Name: crp.Name})
-			return admission.Denied(err.Error())
-		}
 		if req.Operation == admissionv1.Update {
 			var oldCRP placementv1beta1.ClusterResourcePlacement
 			if err := v.decoder.DecodeRaw(req.OldObject, &oldCRP); err != nil {
 				return admission.Errored(http.StatusBadRequest, err)
 			}
+			// this is a special case where we allow updates to old v1beta1 CRP with invalid fields so that we can
+			// update the CRP to remove finalizer then delete CRP.
+			if err := validator.ValidateClusterResourcePlacement(&oldCRP); err != nil {
+				if crp.DeletionTimestamp != nil {
+					return admission.Allowed(allowUpdateOldInvalidCRPFmt)
```

Comment:
in general, do we have enough log to debug? Maybe start at 3+ level so just for e2e